File: Examples/test-suite/ocaml/extend_special_variables_runme.ml
open Swig
open Extend_special_variables

let f = new_ForExtensionNewName '()
let s = f -> "extended_renamed" () as string
let s2 = f -> "extended_renamed" (10) as string

let _ =
  assert (s = "name:extended symname:extended_renamed wrapname: overname:__SWIG_0 decl:ForExtension::extended() fulldecl:char const * ForExtension::extended() parentclasssymname:ForExtensionNewName parentclassname:ForExtension")
  assert (s2 = "name:extended symname:extended_renamed wrapname: overname:__SWIG_1 decl:ForExtension::extended(int) fulldecl:char const * ForExtension::extended(int) parentclasssymname:ForExtensionNewName parentclassname:ForExtension")
;;

let e = new_ExtendTemplateInt '()
let _ = e -> "extending" ()

File: Examples/test-suite/lua/extend_special_variables_runme.lua
local v=require("extend_special_variables")

function check(received, expected)
    assert(received == expected, "Incorrect, received: " .. received)
end

local f = v.ForExtensionNewName()
check(f:extended_renamed(), "name:extended symname:extended_renamed wrapname: overname:__SWIG_0 decl:ForExtension::extended() fulldecl:char const * ForExtension::extended() parentclasssymname:ForExtensionNewName parentclassname:ForExtension")
check(f:extended_renamed(10), "name:extended symname:extended_renamed wrapname: overname:__SWIG_1 decl:ForExtension::extended(int) fulldecl:char const * ForExtension::extended(int) parentclasssymname:ForExtensionNewName parentclassname:ForExtension")
